infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -484,7 +490,29 @@ NetworkMetrics getConnectivityView() {
         // If the management server is not bootstrapped, stamp with INVALID_EPOCH.
         long peerConnectivitySnapshotEpoch = serverContext.getManagementLayout() != null
                 ? serverContext.getManagementLayout().getEpoch() : Layout.INVALID_EPOCH;
-        return new NetworkMetrics(peerConnectivitySnapshotEpoch, peerConnectivityDeltaMap);
+        return new NetworkMetrics(peerConnectivitySnapshotEpoch, ImmutableMap.copyOf(clusterView));
+    }
+
+    /**
+     * Updates the cluster view of this management agent.
+     * From the union of the peer views received in the poll report and the existing peer views
+     * recorded earlier in the cluster view, we retain the latest peer views by comparing the
+     * heartbeat counters.
+     *
+     * @param pollReport Poll Report to update the current cluster view.
+     */
+    private void updatePeerConnectivityMap(PollReport pollReport) {
+        pollReport.getNodeViewMap().values().stream()
+                .map(nodeView -> nodeView.getNetworkMetrics().getPeerConnectivityView())
+                .flatMap(stringPeerViewMap -> stringPeerViewMap.entrySet().stream())
+                .forEach(stringPeerViewEntry -> clusterView
+                        .compute(stringPeerViewEntry.getKey(), (s, peerView)
+                                -> (peerView == null
```

Comment:
we might be missing the case when the peerView for a peer received by an intermediary node however the current node (whose management agent is aggregating) view must take precedence. i.e. current node not receiving the heartbeat response from the peer must lead to updating that edge on the graph (cluster view matrix) regardless of intermediary peerView report.

Reply:
The latest view of any node view takes precedence regardless of what the current node sees. This is the only way the node can aggregate the cluster view and learn about link failures.